packages/story-editor/src/components/library/panes/media/local/hotlink/useInsert.js

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback } from '@web-stories-wp/react';
+import { __ } from '@web-stories-wp/i18n';
+import {
+  getFileExtFromUrl,
+  getFileNameFromUrl,
+  getFirstFrameOfVideo,
+} from '@web-stories-wp/media';
+
+/**
+ * Internal dependencies
+ */
+import { isValidUrl } from '../../../../../../utils/url';
+import useLibrary from '../../../../useLibrary';
+import getResourceFromUrl from '../../../../../../app/media/utils/getResourceFromUrl';
+import {
+  getPosterName,
+  useUploadVideoFrame,
+} from '../../../../../../app/media/utils';
+import { useConfig } from '../../../../../../app/config';
+
+// @todo Get the mime type from server-side validation instead.
+const EXT_MIME_TYPES = {
+  jpg: 'image/jpeg',
+  gif: 'image/gif',
+  jpe: 'image/jpe',
+  jpeg: 'image/jpg',
+  m4v: 'video/mp4',
+  mp4: 'video/mp4',
+  png: 'image/png',
+  webm: 'video/webm',
+  webp: 'image/webp',
+};
+
+function useInsert({
+  link,
+  setLink,
+  errorMsg,
+  setErrorMsg,
+  getFileInfo,
+  onClose,
+}) {
+  const { insertElement } = useLibrary((state) => ({
+    insertElement: state.actions.insertElement,
+  }));
+  const {
+    capabilities: { hasUploadMediaAction },
+  } = useConfig();
+  const { uploadVideoPoster } = useUploadVideoFrame({});
+  const onInsert = useCallback(async () => {
+    if (errorMsg?.length) {
+      return;
+    }
+    const insertionError = __(
+      'Media failed to load. Please ensure the link is valid and the site allows linking from external sites',
```

Comment:
Can we add a period at the end of the string? 

```suggestion
      'Media failed to load. Please ensure the link is valid and the site allows linking from external sites.',
```

Reply:
Sure, but this whole UI will likely undergo many changes anyway.